### PR TITLE
Handle sub-tracks and their titles

### PR DESCRIPTION
**Pull request by zas**
https://github.com/murdos/musicbrainz-userscripts/issues/31

Tested on http://www.discogs.com/release/5880212

### CODE REVIEW

**Comment by murdos** [comment]
Thanks.

Just for the record, http://www.discogs.com/release/5880212 is a wrong example, since this release should use heading tracks instead.
http://www.discogs.com/release/1156598 is a better example.

File: discogs_importer.user.js [hunk 1]
```diff
@@ -386,6 +386,29 @@ function parseDiscogsRelease(data) {
         // Track position and release number
         var trackPosition = discogsTrack.position;
 
+        // Handle sub-tracks
+        if (trackPosition == "" && discogsTrack.sub_tracks) {
+            trackPosition = discogsTrack.sub_tracks[0].position;
```
Comment (murdos) [inline discogs_importer.user.js]: Use first subtrack position is not perfect, but better than nothing.

**Comment by murdos** [comment]
Another improvement could be to compute the track length by summing length of subtracks.

**Comment by zas** [comment]
> Another improvement could be to compute the track length by summing length of subtracks.
http://www.discogs.com/release/5880212 has duration provided in 'index' section, but yes, http://www.discogs.com/release/1156598 may need this feature ;)

File: discogs_importer.user.js (hunk 1 above)
Comment (zas) [inline discogs_importer.user.js]: Yes, this is a bit hacky, but discogs data format doesn't help ;)

Comment (zas) [inline discogs_importer.user.js]: Note: track position isn't correct in all cases,  ie. http://www.discogs.com/release/1156598 will have track B in release editor set as track 5. I found various weird formats in positions like 'AAa' or 'Ab' or 'AA1'... the current script doesn't handle those.

**Pull request merged**